graph/patterns/navigation-property.md

```diff
@@ -0,0 +1,208 @@
+
+`/invoice/{invoiceId}/items/{itemId}?expand=parentInvoice(select=invoiceDate,Customer)`
+
+```xml
+<EntityType Name="invoice">
+  <NavigationProperty Name="items" Type="invoiceItem" ContainsTarget="true" >
+</EntityType>
+
+<EntityType Name="invoiceItem">
+  <NavigationProperty Name="parentInvoice" Type="invoice" ContainsTarget="false" >
+</EntityType>
+```
+
+One-to-many relationships may be contained or non-contained relations.
+
+
+## Example
+-------
+
+### Retrieving a related entity
+
+```http
+GET /users/{id}/manager
+
+200 OK
+Content-Type: application/json
+
+{
+  "id": "6b3ee805-c449-46a8-aac8-8ff9cff5d213",
+  "displayName": "Bob Boyce"
+}
+```
+
+This navigation property could be described with the following CSDL: 
+```xml
+<EntityType Name="user">
+  <NavigationProperty Name="manager" Type="graph.user" ContainsTarget="false" >
+</EntityType>
+```
+`ContainsTarget` is set to false for clarity, this is the default value when the attribute is omitted.
+### Retrieving a reference to a related entity
+
+```http
+GET /users/{id}/manager/$ref
+
+200 OK
+Content-Type: application/json
+
+{
+  "@odata.id": "https://graph.microsoft.com/v1.0/directoryObjects/6b3ee805-c449-46a8-aac8-8ff9cff5d213/Microsoft.DirectoryServices.User"
+}
+```
+Note: Currently the base URL returned in $ref results are incorrect.  In order to process these URLs the client will need to convert the URL to a Graph URL.
+
+### Retrieving an entity with a related entity included
+
+```http
+GET /users/{id}?select=id,displayName&expand=manager(select=id,displayName)
+
+200 OK
+Content-Type: application/json
+
+{
+  "id": "3f057904-f936-4bf0-9fcc-c1e6f84289d8",
+  "displayName": "Jim James",
+  "manager": {
+    "@odata.type": "#microsoft.graph.user",
+    "id": "6b3ee805-c449-46a8-aac8-8ff9cff5d213",
+    "displayName": "Bob Boyce"
+  }
+}
+```
+
+### Creating an entity with a reference to a related entity
+
+Create a new user that references an existing manager
+```http
+POST /users
+Content-Type: application/json
+
+{
+    "displayName": "Bob",
+    "manager@odata.bind": "https://graph.microsoft.com/v1.0/users/{managerId}"
+}
+
+201 Created
+```
+
+### Updating a related entity reference
+
+Update the user entity to contain a relationship to an existing manager.
+ 
+```http
+PATCH /users/{id}
+Content-Type: application/json
+
+{
+    "displayName": "Bob",
+    "manager@odata.bind": "https://graph.microsoft.com/v1.0/users/{managerId}"
+}
+
+204 No Content
+```
+
+Create a relationship between the user and the existing manager.
+
+```http
+PUT /users/{id}/manager/$ref
```

Comment:
If we have two ways of achieving the same result what is our preference? What do we ask from our workloads to implement one or both?

Reply:
Removed the example that requires supporting the $ref endpoint.